aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -113,4 +133,29 @@ void aesd_circular_buffer_init(struct aesd_circular_buffer *buffer)
     memset(buffer,0,sizeof(struct aesd_circular_buffer));
 }
 
+/**
+* free the circular buffer described by @param buffer to delete each entry
+*/
+void aesd_circular_buffer_delete(struct aesd_circular_buffer *buffer)
+{
+	uint8_t index;
+	struct aesd_buffer_entry *entry;
+	AESD_CIRCULAR_BUFFER_FOREACH(entry, buffer, index) 
```

Comment:
@MuktaDarekar I'd move the logic for deleting entries into the application, since it will know whether `buffptr` was dynamically allocated.